daemon/api_validate.go

```diff
@@ -0,0 +1,260 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/overlord/assertstate"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+type validationSetResult struct {
+	ValidationSet string `json:"validation-set,omitempty"`
+	Mode          string `json:"mode"`
+	Seq           int    `json:"seq,omitempty"`
+	Valid         bool   `json:"valid"`
+	// TODO: attributes for Notes column
+}
+
+func modeString(mode assertstate.ValidationSetMode) (string, error) {
+	switch mode {
+	case assertstate.Monitor:
+		return "monitor", nil
+	case assertstate.Enforce:
+		return "enforce", nil
+	}
+	return "", fmt.Errorf("internal error: unhandled mode %d", mode)
+}
+
+func validationSetKeyString(accountID, name string, sequence int) string {
+	key := assertstate.ValidationSetKey(accountID, name)
+	if sequence != 0 {
+		key = fmt.Sprintf("%s=%d", key, sequence)
+	}
+	return key
+}
+
+func validationSetNotFound(accountID, name string, sequence int) Response {
+	res := &errorResult{
+		Message: "validation set not found",
+		Kind:    client.ErrorKindValidationSetNotFound,
+		Value:   validationSetKeyString(accountID, name, sequence),
```

Comment:
we need to think a bit about this, should it be a string? or some kind of maps with exploded key components?

Reply:
I agree, I think it makes sense to make life easier for the clients.